tools/c7n_azure/c7n_azure/provider.py

```diff
@@ -43,7 +60,29 @@ def initialize_policies(self, policy_collection, options):
     def get_session_factory(self, options):
         return partial(Session,
                        subscription_id=options.account_id,
-                       authorization_file=options.authorization_file)
+                       authorization_file=options.authorization_file,
+                       cloud_endpoints=self.cloud_endpoints)
+
+    def _get_cloud_endpoints(self, options):
+        cloud_list = options.get('regions')
+
+        if not cloud_list:
+            return AZURE_PUBLIC_CLOUD
+        elif len(cloud_list) > 1:
+            log.error('Multiple Azure Clouds provided. Please pass in only one.')
+            sys.exit(1)
+
+        # Only support passing in one cloud at a time
+        cloud = self.region_to_cloud.get(cloud_list[0])
+
+        if cloud:
+            return cloud
+        else:
+            log.error('Region Flag: %s not recognized, please choose an Azure Cloud from'
+                      'the following: AzureCloud, AzureChinaCloud, AzureGermanyCloud, '
+                      'AzureUSGovernment'
```

Comment:
If you want to avoid keeping two lists of cloud names, `", ".join(region_to_cloud.keys())`